tests/integration_test/shadow_system_test.c

```diff
@@ -0,0 +1,934 @@
+        TEST_ASSERT_FALSE( receivedPubAck );
+    }
+
+    return mqttStatus;
+}
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void shadowTestSetUp()
+{
+    /* Reset file-scoped global variables. */
+    receivedSubAck = false;
+    receivedUnsubAck = false;
+    receivedPubAck = false;
+    receivedPubRec = false;
+    receivedPubRel = false;
+    receivedPubComp = false;
+    persistentSession = false;
+    receivedUpdateDeltaResult = false;
+    receivedUpdateDocumentsResult = false;
+    receivedUpdateAcceptedResult = false;
+    receivedUpdateRejectedResult = false;
+    receivedDeleteAcceptedResult = false;
+    receivedDeleteRejectedResult = false;
+    receivedGetAcceptedResult = false;
+    receivedGetRejectedResult = false;
+
+    memset( &incomingInfo, 0u, sizeof( MQTTPublishInfo_t ) );
+
+    /* Initializer server information. */
+    serverInfo.pHostName = BROKER_ENDPOINT;
+    serverInfo.hostNameLength = strlen( BROKER_ENDPOINT );
+    serverInfo.port = BROKER_PORT;
+
+    /* Initialize SocketsConfig. */
+    socketsConfig.enableTls = true;
+    socketsConfig.pAlpnProtos = NULL;
+    socketsConfig.maxFragmentLength = 0;
+    socketsConfig.disableSni = true;
+    socketsConfig.pRootCa = SERVER_ROOT_CA_CERT;
+    socketsConfig.rootCaSize = strlen( SERVER_ROOT_CA_CERT ) + 1U;
+    socketsConfig.sendTimeoutMs = TRANSPORT_SEND_RECV_TIMEOUT_MS;
+    socketsConfig.recvTimeoutMs = TRANSPORT_SEND_RECV_TIMEOUT_MS;
+
+    /* Establish a TCP connection with the server endpoint, then
+     * establish TLS session on top of TCP connection. */
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_SUCCESS, SecureSocketsTransport_Connect( &networkContext,
+                                                                                        &serverInfo,
+                                                                                        &socketsConfig ) );
+
+    /* Establish MQTT session on top of the TCP+TLS connection. */
+    establishMqttSession( &context, &networkContext, true, &persistentSession );
+}
+
+/* Called after each test method. */
+void shadowTestTearDown()
+{
+    /* Free memory, if allocated during test case execution. */
+    if( incomingInfo.pTopicName != NULL )
+    {
+        free( ( void * ) incomingInfo.pTopicName );
+    }
+
+    if( incomingInfo.pPayload != NULL )
+    {
+        free( ( void * ) incomingInfo.pPayload );
+    }
+
+    /* Terminate MQTT connection. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Disconnect( &context ) );
+
+    /* Terminate TLS session and TCP connection. */
+    ( void ) SecureSocketsTransport_Disconnect( &networkContext );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test group for device Shadow system tests.
+ */
+TEST_GROUP( deviceShadow_Integration );
+
+TEST_SETUP( deviceShadow_Integration )
+{
+    shadowTestSetUp();
```

Comment:
All the logic of `shadowTestSetUp` and `shadowTestTearDown` can be placed within the `TEST_SETUP` and `TEST_TEAR_DOWN` macros respectively. The MQTT integration tests separated them as they dealt with 2 different test groups (`coreMQTT_Integration` and `coreMQTT_AWS_IoT_Compatible`), and thus, the functions were added to commonize the logic to use in both the test groups. 
As Shadow only has a single test group, these functions don't need to be created

Reply:
sure, will modify it.